src/pretix/base/models/items.py

```diff
@@ -897,3 +897,31 @@ def _position_lookup(self) -> Q:
 
     class QuotaExceededException(Exception):
         pass
+
+    @staticmethod
+    def clean_variations(items, variations):
+        for variation in variations:
+            if variation.item not in items:
+                raise ValidationError(_('All variations must belong to an item contained in the items list'))
+                break
+
+    @staticmethod
+    def clean_items(event, items, variations):
+        for item in items:
+            if event != item.event:
+                raise ValidationError(_('One or more items does not belong to this event'))
+            if item.has_variations:
+                selected_vars = set(item.variations.all()).intersection(variations)
+                if bool(selected_vars) is False:
```

Comment:
I find this hard to read and understand. Also, the query to item.variations.all() is not really required. How about:

``
if not any(var.item == item for var in variations):
``

Reply:
Much better - thanks. Still getting up to speed in python. Didn't know that construct.